src/lib/server/cart.ts

```diff
@@ -114,7 +115,20 @@ export async function addToCartInDb(
 	) {
 		throw error(400, 'Cart has too many items');
 	}
-
+	const variationPriceArray = params.chosenVariations
+		? Object.entries(params.chosenVariations).map((variation) => ({
+				amount:
+					product.variations?.find(
+						(vari) => variation[0] === vari.name && variation[1] === vari.value
+					)?.price ?? 0,
+				currency: runtimeConfig.mainCurrency
+		  }))
+		: [];
```

Comment:
Please add a check that if a product has variations in DB, a variation for each kind should be chosen by the customer (and not more than one)

also the same check in `createOrder` (in case the variations of the product changed between adding to cart & making order)